tests/test_arrow_dataset.py

```diff
@@ -3898,12 +3902,13 @@ def test_dataset_from_generator_keep_in_memory(keep_in_memory, data_generator, t
 def test_dataset_from_generator_features(features, data_generator, tmp_path):
     cache_dir = tmp_path / "cache"
     default_expected_features = {"col_1": "string", "col_2": "int64", "col_3": "float64"}
+    split = NamedSplit("validation")
     expected_features = features.copy() if features else default_expected_features
     features = (
         Features({feature: Value(dtype) for feature, dtype in features.items()}) if features is not None else None
     )
-    dataset = Dataset.from_generator(data_generator, features=features, cache_dir=cache_dir)
-    _check_generator_dataset(dataset, expected_features)
+    dataset = Dataset.from_generator(data_generator, features=features, cache_dir=cache_dir, split=split)
+    _check_generator_dataset(dataset, expected_features, split)
 
```

Comment:
I would add a specific `test_dataset_from_generator_split` with a parametrized `split` values, such as not passing any value, passing `NamedSplit("train")`, passing literal `"train"`, passing other NamedSplit, etc.

Reply:
`test_dataset_from_generator_split` added, still i have impacted `_check_generator_dataset` to share the same generic check everywhere